aries_cloudagent/anoncreds/holder.py

```diff
@@ -259,6 +279,58 @@ async def store_credential(
 
         return credential_id
 
+    async def store_credential_w3c(
+        self,
+        credential_definition: dict,
+        credential_data: dict,
+        credential_request_metadata: dict,
+        credential_attr_mime_types: dict = None,
+        credential_id: str = None,
+        rev_reg_def: dict = None,
+    ) -> str:
+        """Store a credential in the wallet.
+
+        Args:
+            credential_definition: Credential definition for this credential
+            credential_data: Credential data generated by the issuer
+            credential_request_metadata: credential request metadata generated
+                by the issuer
+            credential_attr_mime_types: dict mapping attribute names to (optional)
+                MIME types to store as non-secret record, if specified
+            credential_id: optionally override the stored credential id
+            rev_reg_def: revocation registry definition in json
+
+        Returns:
+            the ID of the stored credential
+
+        """
+        try:
+            secret = await self.get_master_secret()
+            cred_w3c = W3cCredential.load(credential_data)
+            cred_w3c.to_legacy()
+            await asyncio.get_event_loop().run_in_executor(
+                None,
+                cred_w3c.process,
+                credential_request_metadata,
+                secret,
+                credential_definition,
+                rev_reg_def,
+            )
+            # TODO we want to store the credential in the W3C format in the wallet,
+            # This will require changes to other endpoints that fetch credentials
+            cred_recvd = Credential.from_w3c(cred_w3c)
```

Comment:
Need to call the process method on this legacy credential, something like:

```
            cred_legacy = Credential.from_w3c(cred_w3c)
            cred_recvd = await asyncio.get_event_loop().run_in_executor(
                None,
                cred_legacy.process,
                credential_request_metadata,
                secret,
                credential_definition,
                rev_reg_def,
            )
```

(this is why the legacy proof request is failing on the vc_di-issued credential)

Reply:
ok my understanding this is implemented, so we can resolve?

Reply:
It's not resolved, the above comments still needs to be addressed.  If you run the demo, issue a vc_di credential, and then request an indy (legacy) proof, the proof fails, because the `cred_legacy.process` method is never called.  (Or it's now called `cred_recd` in the code actually)

Reply:
Since the underlying credential we are issuing is an indy credential (even though it's issued in vc_di format) we should be able to use this credential in both the legacy and new dif proof requests.  (Likewise a legacy indy credential should work in a new dif proof request.)